activerecord/test/cases/enum_test.rb

```diff
@@ -572,7 +592,7 @@ def self.name; "Book"; end
       e = assert_raises(ArgumentError, "enum value `#{value}` should not be allowed") do
         Class.new(ActiveRecord::Base) do
           self.table_name = "books"
-          enum category: [:other, value]
```

Comment:
I had to change some tests because now enums using array need `declared attribute` at definition time raising this exception before those evaluated in the tests.